python/insights_ansible_playbook_lib/_keygen.py

```diff
@@ -0,0 +1,214 @@
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = "gpg-"
+else:
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY = "/tmp/"
+    TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX = (
+        "insights-ansible-playbook-verifier-gpg-"
+    )
+
+
+def _run_gpg_command(command: list[str], fingerprint: bool = False) -> GPGCommandResult:
+    """Run a GPG command in a specific directory.
+
+    :param command: The command to be executed.
+    :param fingerprint: Optional flag to get the fingerprint from the output.
+    """
+    process = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        env={"LC_ALL": "C.UTF-8"},
+        text=fingerprint,
+    )
+    stdout, stderr = process.communicate()
+
+    result = GPGCommandResult(
+        ok=process.returncode == 0,
+        return_code=process.returncode,
+        stdout=stdout.decode("utf-8") if not fingerprint else stdout,
+        stderr=stderr.decode("utf-8") if not fingerprint else stderr,
+        _command=None,
+    )
+
+    if result.ok:
+        logger.debug(f"GPG command {command}: ok.")
+    else:
+        logger.error(f"GPG command {command} returned non-zero code: {result}.")
+
+    return result
+
+
+def _generate_keys() -> str:
+    """
+    Generate GPG keys into a temporary directory.
+    """
+    # Create a temporary directory to store the keys
+    gpg_tmp_dir = tempfile.mkdtemp(
+        dir=TEMPORARY_GPG_HOME_PARENT_DIRECTORY,
+        prefix=TEMPORARY_GPG_HOME_PARENT_DIRECTORY_PREFIX,
+    )
+    logger.debug(f"Generating GPG keys into {gpg_tmp_dir}.")
+
+    instructions_file = pathlib.Path(gpg_tmp_dir) / "keygen"
+    instructions_file.write_text(
+        """Key-Type: EDDSA
+Key-Curve: ed25519
+Subkey-Type: ECDH
+Subkey-Curve: cv25519
+Name-Real: insights-ansible-playbook-verifier test
+Expire-Date: 0
+%no-protection
+%commit
+"""
+    )
+    instructions_text = instructions_file.read_text().strip().replace("\n", "\\n")
+    logger.debug(
+        f"Keys generation instructions written to file {instructions_file}: {instructions_text}"
+    )
+
+    # Generate the keys in a temporary directory
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--batch",
+            "--homedir",
+            gpg_tmp_dir,
+            "--generate-key",
+            f"{instructions_file}",
+        ]
+    )
+
+    return gpg_tmp_dir
+
+
+def _export_key_pair(gpg_tmp_dir: str, keys_path: str) -> None:
+    """
+    Export the generated key pair to the `keys_path` directory.
+
+    :param gpg_tmp_dir: The GPG home directory where the key pair was generated.
+    :param keys_path: The directory where the key pair should be exported.
+    """
+    _run_gpg_command(
+        [
+            "/usr/bin/gpg",
+            "--homedir",
+            gpg_tmp_dir,
+            "--export",
+            "--armor",
+            "--yes",
+            "--output",
+            f"{keys_path}/key.public.gpg",
+        ]
+    )
+
```

Comment:
Should we `logger.debug()` that the key was saved into path ...? It would be consistent with the fingerprint.

Non-blocking.